xonsh/completers/completer.py

```diff
@@ -125,3 +190,41 @@ def remove_completer(name: str):
         return
     else:
         return None, err + "\n", 1
+
+
+class CompleterAlias(ArgParserAlias):
+    """CLI to add/remove/list xonsh auto-complete functions"""
+
+    def build(self):
+        parser = self.create_parser(prog="completer")
+        parser.add_command(_register_completer, prog="add")
+        parser.add_command(remove_completer, prog="remove", aliases=["rm"])
+        parser.add_command(list_completers, prog="list", aliases=["ls"])
+        return parser
+
+
+completer_alias = CompleterAlias()
+
+
+@contextual_command_completer
+def complete_argparser_aliases(command: CommandContext):
```

Comment:
I decided to break this PR. in the upcoming PR I will keep this inside completer/_alias.py . I just wanted to merge all things related to completer alias in one place. What do you think about it? which will be better, I don't like the preceding underscore. It has some public functions, like the ones to remove/add completers.

Reply:
I'm fine with creating a new `aliases.py` file for this function, since it **is** the completer for aliases :)
I rather you do it in this PR if you don't mind, generally I intend on releasing versions more frequently and I prefer inserting it correctly in the first place rather than inserting and then refactoring

Reply:
I'm continuing the threads here instead of re-opening them in https://github.com/xonsh/xonsh/pull/4391

Reply:
Also `ArgparseCompleter` should be there

Reply:
> Also `ArgparseCompleter` should be there

it is now in the `completer/tools.py` along other completion helpers.

Reply:
It isn't a tool for other completera to use, it's the completion logic for aliases. 
Just like the bash completion logic is in `bash.py`, and completion logic for paths is in `paths.py`, the completion logic for aliases needs to be in an `aliases.py` file.
So goes for all the other logic you put in `completer.py`

Reply:
moved `complete_argparser_aliases` to completers/_aliases.py  in 
 
https://github.com/xonsh/xonsh/pull/4391/commits/2e45190436ebdd52e874f28142e3dbea20db05a3

Reply:
Same for `ArgparseCompleter` and `complete_argparser`
Sorry I'm still pressing this it's just that this bloats `completers/tools.py` with things that are specific to the aliases completer only

Reply:
the completion logic is for argparse.Parser. For some external command, one can build a completer from this class. This is not specific to alias.

Reply:
anyways I will move this class to `cli_utils.py` if you are concerned.

Reply:
I have created its own module inside completers.

Reply:
@daniel-shimon I think this is the final thing to resolve. Do I need to update the amalgamate setting now?

Reply:
This can be resolved now :)
Go ahead!